mesonbuild/wrap/wrap.py

```diff
@@ -600,6 +600,9 @@ def _get_git(self, packagename: str) -> None:
         if self.wrap.values.get('depth', '') != '':
             is_shallow = True
             depth_option = ['--depth', self.wrap.values.get('depth')]
+        clone_recursive_val = self.wrap.values.get('clone-recursive', '').lower()
```

Comment:
Placing this earlier in the function is an improvement. Now if there's an invalid value in the wrap file, meson will error out sooner. When the value is corrected, meson will clone the repository and the submodules.

Previously, after a failed attempt, meson would make no attempt to clone the submodules even after the value was corrected in the wrap file.

There is still a parser error if no value is provided to clone-recursive, which is the desired behavior. An exception will be raised if a value is given other than 'true' or 'false'. If 'clone-recursive' is not included in the wrap file at all, 'false' will be assumed.